### ISSUE TITLE
Docker compose doesn't work with begriffs/postgrest:latest

### ISSUE THREAD

**Issue by tonyhschu** [begriffs/postgrest#868]
... but does work with begriffs/postgrest:v0.4.0.0

To reproduce, clone the repo to the latest master, navigate to `/docker/` then run:

`$ docker-compose up`

Visiting localhost:3000 you'll see:

```json
{
    "details":"could not connect to server: No such file or directory\n\tIs the server running locally and accepting\n\tconnections on Unix domain socket \"/var/run/postgresql/.s.PGSQL.5432\"?\n",
    "code":"","message":"Connection error"
}
```

By changing the image used in the docker-compose.yml file to `begriffs/postgrest:v0.4.0.0` I was able to get it up and running.

Tony

**Comment by ruslantalpa** [comment]
try the second time with 4.0, i bet it will fail

**Comment by ajnsit** [comment]
I'm not sure if this is the same issue or not, but the one of the problems I faced was that postgrest would run before postgres had started accepting connections. The solution was to wrap postgrest in a script that checks for postgres being up before running postgrest.

**Comment by ruslantalpa** [comment]
Yes, it's that. And there is a new PR now to fix this :)

**Comment by mattddowney** [comment]
Quick fix for this is to just add `restart: always` to the container

**Comment by begriffs** [comment]
0.4.1.0 will have the fix in it that will retry the connection as needed.

**Comment by tonyhschu** [comment]
Is there a flag in the docker-compose file I should be using to enable this retry behavior? I've upgraded to using 0.4.1.0, but Postgrest continues to exit after failing to connect after the first try.

**Comment by begriffs** [comment]
The connection retrying functionality is on the master branch but as yet [unreleased](https://github.com/begriffs/postgrest/blob/master/CHANGELOG.md#unreleased) in a new version.

**Comment by mattddowney** [comment]
Yes. Add `restart: always` in the compose file like [here](https://github.com/mattddowney/compose-postgrest/blob/master/docker-compose.yml#L46)

**Comment by ruslantalpa** [comment]
this auto reconnect feature was implemented and is in 4.2 already i think.
